relentless/optimize/method.py

```diff
@@ -280,57 +351,116 @@ def max_iter(self, value):
 class SteepestDescent(Optimizer):
     r"""Steepest descent algorithm.
 
-    For an :class:`~relentless.optimize.ObjectiveFunction` :math:`f\left(\mathbf{x}\right)`,
+    For an :class:`~relentless.optimize.objective.ObjectiveFunction` :math:`f\left(\mathbf{x}\right)`,
     the steepest descent algorithm seeks to approach a minimum of the function.
 
-    With an initial numerical guess for all the design variables :math:`\mathbf{x}`,
-    the following iterative calculation is performed:
+    :math:`\mathbf{X}=\left[X_1,\ldots,X_n\right]` is defined as the scaling
```

Comment:
I think alot of the same comments will apply here, so I am going to stop reviewing for now. Let's get feedback from Zach and resolve what's going on in the LineSearch, apply some of the simplifying changes I proposed above to both classes, and then I'll take a look at this again!